test_communication/package.xml

```diff
@@ -11,21 +11,24 @@
   <maintainer email="dthomas@osrfoundation.org">Dirk Thomas</maintainer>
   <license>Apache License 2.0</license>
 
+  <build_depend>ament_cmake_auto</build_depend>
+  <build_depend>builtin_interfaces</build_depend>
+  <build_depend>rosidl_default_generators</build_depend>
+
   <buildtool_depend>ament_cmake</buildtool_depend>
```

Comment:
`ament_cmake_auto` aims to automate several steps in your CMake and you need to call its function in order to actually perform all those: see https://github.com/ament/ament_cmake/blob/master/ament_cmake_auto/cmake/ament_auto_package.cmake You either don't need any of these (than you probably want to use `ament_cmake` instead) or you haven't noticed that they are not happening.